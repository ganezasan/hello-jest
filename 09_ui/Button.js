import React, { useState } from 'react';

export default function Button({ isToggle = true }) {
  const [isToggleOn, setIsToggleOn] = useState(isToggle);

  const buttonStyle = {
    backgroundColor: 'green',
    color: 'white'
  };

  return (
    <button onClick={() => setIsToggleOn(!isToggleOn)} style={buttonStyle}>
      { isToggleOn ? 'ON' : 'OFF' }
    </button>
  );
}
