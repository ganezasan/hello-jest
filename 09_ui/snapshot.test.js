import React from 'react';
import renderer from 'react-test-renderer';
import { render, fireEvent } from '@testing-library/react';
import Button from './Button';

describe('snapshot test', () => {
  it('renders correctly with react-test-renderer', () => {
    const button = renderer.create(<Button />);

    let tree = button.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with @testing-library/react', () => {
    const button = render(<Button />);
    expect(button.container).toMatchSnapshot();
    
    // manually click the button, then its text will update to OFF
    fireEvent.click(button.getByText('ON'));
    expect(button.container).toMatchSnapshot();
    
    // manually click the button, then its text will update to ON
    fireEvent.click(button.getByText('OFF'));
    expect(button.container).toMatchSnapshot();
  }); 
});