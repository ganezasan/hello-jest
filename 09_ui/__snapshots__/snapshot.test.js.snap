// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test renders correctly with @testing-library/react 1`] = `
<div>
  <button
    style="background-color: green; color: white;"
  >
    ON
  </button>
</div>
`;

exports[`snapshot test renders correctly with @testing-library/react 2`] = `
<div>
  <button
    style="background-color: green; color: white;"
  >
    OFF
  </button>
</div>
`;

exports[`snapshot test renders correctly with @testing-library/react 3`] = `
<div>
  <button
    style="background-color: green; color: white;"
  >
    ON
  </button>
</div>
`;

exports[`snapshot test renders correctly with react-test-renderer 1`] = `
<button
  onClick={[Function]}
  style={
    Object {
      "backgroundColor": "green",
      "color": "white",
    }
  }
>
  ON
</button>
`;
